src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,70 @@
+package mate.academy.springboot.datajpa.controller;
+
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import mate.academy.springboot.datajpa.mapper.DtoMapper;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.model.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.model.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+@AllArgsConstructor
+public class ProductController {
+    private final ProductService productService;
+    private final DtoMapper<ProductRequestDto, ProductResponseDto, Product> mapper;
+
+    @PostMapping
+    public ProductResponseDto save(@RequestBody @Validated ProductRequestDto categoryRequestDto) {
+        Product product = productService.save(mapper.mapToModel(categoryRequestDto));
+        return mapper.mapToDto(product);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto get(@PathVariable Long id) {
+        return mapper.mapToDto(productService.get(id));
+    }
+
+    @GetMapping("/by-price")
+    public List<ProductResponseDto> getAllByPrice(@RequestParam @Positive BigDecimal from,
+                                                  @RequestParam @Positive BigDecimal to) {
+        return productService.findAllByPriceBetween(from, to).stream()
+                .map(mapper::mapToDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/by-categories")
+    public List<ProductResponseDto> getAllByCategories(@RequestParam String[] categoriesId) {
```

Comment:
Use plural in the end of the variable names instead of in the middle, e.g. `categoryIds` or `categoryNames`. 
Please fix the similar cases in the code.